pkg/provider/api.go

```diff
@@ -316,6 +318,14 @@ func newWithXOXP(transport string, authProvider auth.ValueAuth, logger *zap.Logg
 		err    error
 	)
 
+	var redisClient *RedisClient
+	if os.Getenv("REDIS_ADDR") != "" || os.Getenv("REDIS_PASSWORD") != "" || os.Getenv("REDIS_DB") != "" {
```

Comment:
We'll need to set these env vars in the cloud run service - not sure if we want to specify env vars per mcp server, or just add the redis ones to the general env var list [here](https://github.com/natomalabs/natoma/blob/main/packages/server/src/lib/mcp/utils.ts#L258)